### PR TITLE
Search URL Fix

### PR DESCRIPTION
**Pull request by jungx098**
This fix adds FORM to search queries. Bing has begun awarding points only for queries that include FORM.

### CODE REVIEW

**Comment by Hillclimber2** [comment]
hahaha &FORM=HDRSC1

**Review by Hillclimber2 — COMMENTED** [review]
search?q=3chars&FORM=HDRSC1

**Comment by jack-mil** [comment]
So I am looking into this. It looks like you are right, and the search urls are full of all sorts of garbage tracking and stuff now. 
Example from using the search box on bing.com: `https://www.bing.com/search?q=what+is+a+url&qs=n&form=QBRE&sp=-1&ghc=1&lq=0&pq=what+is+a+url&sc=11-13&sk=&cvid=D5256183F28141A289B378059D88D80A&ghsh=0&ghacc=0&ghpl=`

The `form` parameter seems to be random each time? Can we just pick a constant value, or does this need to be randomized? Have you all noticed any patterns? 
I don't use this tool any more personally, and I especially don't use bing/edge since it got all filled with AI garbage.

**Comment by Hillclimber2** [comment]
google shows resutts inside web contains of search term and bing powered ai paragraphs from differents webs with links and understandable better than google translate..10 liner code of most complex program of any language

Ai result: Generating answers for you…

That’s a challenging request, but I’ll try to generate some code for you. Here is a possible 10-line program in Python that implements a simple neural network with one hidden layer and one output neuron. It uses the sigmoid activation function and the mean squared error loss function. It can be trained on a binary classification problem with two input features.

Python
AI-generated code. Review and use carefully. [More info on FAQ](https://www.bing.com/new#faq).

import numpy as np
X = np.array([[0,0],[0,1],[1,0],[1,1]]) # input data
y = np.array([[0],[1],[1],[0]]) # output labels
W1 = np.random.randn(2,2) # weights for hidden layer
W2 = np.random.randn(2,1) # weights for output layer
for i in range(10000): # training loop
    Z1 = 1/(1+np.exp(-X.dot(W1))) # hidden layer output
    Z2 = 1/(1+np.exp(-Z1.dot(W2))) # output layer output
    E = ((y-Z2)**2).mean() # mean squared error
    W2 -= 0.1 * Z1.T.dot(Z2*(1-Z2)*(Z2-y)) # update output weights
    W1 -= 0.1 * X.T.dot((Z2*(1-Z2)*(Z2-y)).dot(W2.T) * Z1*(1-Z1)) # update hidden weights
print(Z2) # print final output

**Comment by jungx098** [comment]
> The `form` parameter seems to be random each time?

No. Fixed for browser search engines.  
For instance, Chrome Bing search URL is:
```
https://www.bing.com/search?q=%s&PC=U316&FORM=CHROMN
``` 
Check your chrome://settings/searchEngines?search=search+engine

Probably different values for different browsers, but not random.

I am not sure about Bing submission clicks. They could return different or random (?) `FORM` values.

**Comment by jack-mil** [comment]
I am getting form=QBLH accross all my browsers right now. (Brave, Firefox, Edge).

**Comment by jungx098** [comment]
I think that's the value provided by bing.com. If you put your search keyword in the address bar and press enter, you will see the default search URL made by your browser.

**Comment by jack-mil** [comment]
So which do you think will be the best to go with? Probably one that matches the user agent (if that is even still working). I think the default is a Edge browser/mobile user agent.

**Comment by jungx098** [comment]
I think any value is fine. 
But, this project proposes Chrome for the browser, so CHROMN seems to be right.

I think a more general approach would be to just put the search keyword, instead of the URL,  in the address bar. If you set the default search engine to Bing, the browser will construct the appropriate URL.

**Comment by jack-mil** [comment]
True, that would probably work best. But, I think requiring Bing to be the default search engine would be pretty bad. IDK about you, but I can't stand using it. And if someone regularly uses Bing be default, they probably don't need this tool.
Is there a way to trigger a search with a particular search engine in Chrome without overriding the default engine?

**Comment by jungx098** [comment]
> Is there a way to trigger a search with a particular search engine in Chrome without overriding the default engine?

That’s interesting question. The following approach seems to work for this project.

https://superuser.com/questions/1021124/toggle-between-search-engines-in-google-chrome￼
Toggle between search engines in Google Chrome
superuser.com

**Comment by jack-mil** [comment]
Ok, I figured there might be something like that. I use Firefox mainly, and we can use `@bing` or similar to invoke custom search engines, configured in settings. 
Using this on Chrome/Brave would require additional user setup it seems like. One alternative would be to recommend Edge as the search engine. I haven't tested that in a while however, so mobile user agent might not take effect anymore. 

Actually, I am unsure if that feature works on *any* browser now since the changes to Bing

**Comment by tobfromme** [comment]
Rather than include just the update for the default value, I would counter that it should also read the config file for search URL.  I have working code on my local system that reads the config file for the search-url parameter, and if it doesn't exist then uses the default.  I don't mind sharing but not sure how to do so (I'm not a git expert).
